jsonschema/cli.py

```diff
@@ -1,14 +1,107 @@
 """
 The ``jsonschema`` command line.
 """
+
 from __future__ import absolute_import
+from abc import ABCMeta
 import argparse
 import json
 import sys
 
 from jsonschema import __version__
 from jsonschema._reflect import namedAny
 from jsonschema.validators import validator_for
+from jsonschema.exceptions import SchemaError, ValidationError
+
+
+class _CliOutputWriter():
```

Comment:
Thanks! This was close to what I meant, but there isn't really any benefit to the abstract base class (or the inheritance in general) -- the subclasses can just live on their own without any relationship in the inheritance hierarchy.

Reply:
Indeed there is not in a functional point of view. But I think it helps to enforce a common interface to all of these classes, and it avoids some duplicate code (like CONSTANTS or common initialization). As in the future a new class will be added to print a `json` report, I think this architecture will be even more useful.

Maybe it is a personal preference. But if you think it harms more than it helps, I’ll remove it. You tell me.

Reply:
Your code, your decision :blush: :+1: